sdk/identity/azure-identity/azure/identity/_credentials/service_fabric.py

```diff
@@ -0,0 +1,79 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+import os
+from typing import TYPE_CHECKING
+
+from azure.core.pipeline.transport import HttpRequest
+
+from .. import CredentialUnavailableError
+from .._constants import EnvironmentVariables
+from .._internal.managed_identity_client import ManagedIdentityClient
+from .._internal.get_token_mixin import GetTokenMixin
+
+if TYPE_CHECKING:
+    from typing import Any, Optional
+    from azure.core.credentials import AccessToken
+
+
+class ServiceFabricCredential(GetTokenMixin):
+    def __init__(self, **kwargs):
+        # type: (**Any) -> None
+        super(ServiceFabricCredential, self).__init__()
+
+        client_args = _get_client_args(**kwargs)
+        if client_args:
+            self._client = ManagedIdentityClient(**client_args)
+        else:
+            self._client = None
+
+    def get_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        if not self._client:
+            raise CredentialUnavailableError(
+                message="Service Fabric managed identity configuration not found in environment"
+            )
+        return super(ServiceFabricCredential, self).get_token(*scopes, **kwargs)
+
+    def _acquire_token_silently(self, *scopes):
+        # type: (*str) -> Optional[AccessToken]
+        return self._client.get_cached_token(*scopes)
+
+    def _request_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        return self._client.request_token(*scopes, **kwargs)
+
+
+def _get_client_args(**kwargs):
+    # type: (**Any) -> Optional[dict]
+    identity_config = kwargs.pop("_identity_config", None) or {}
+
+    url = os.environ.get(EnvironmentVariables.IDENTITY_ENDPOINT)
+    secret = os.environ.get(EnvironmentVariables.IDENTITY_HEADER)
+    thumbprint = os.environ.get(EnvironmentVariables.IDENTITY_SERVER_THUMBPRINT)
+    if url and secret and thumbprint:
+        version = "2019-07-01-preview"
+        base_headers = {"Secret": secret}
+        connection_verify = False
+        if kwargs.get("client_id"):
+            identity_config["clientid"] = kwargs.pop("client_id")
```

Comment:
I was under the impression that users would create a `ManagedIdentityCredential` and use that directly, which I thought would make it necessary to handle this here. If that's not the case I can remove this, but I noticed that `AppServiceCredential` [also does this](https://github.com/Azure/azure-sdk-for-python/blob/5d87d7af17ae67a66c3a52d659c9c036bcf5bc12/sdk/identity/azure-identity/azure/identity/_credentials/app_service.py#L69) -- is there a particular reason why?

Reply:
Users write `ManagedIdentityCredential(client_id="...")` and internally we map that to a parameter for the platform's managed identity API. For all but one API the parameter is "client_id", hence ManagedIdentityClient's default behavior. The special case is App Service 2017-09-01, which requires "clientid" instead. So we want these two lines iff the Service Fabric API requires "clientid".

Reply:
I see, I didn't realize that some APIs used `client_id` instead of that one's `clientid`. When we spoke to the service team they said we should just pass along any client ID the user provides but didn't specify how exactly it should be represented